src/twisted/conch/ssh/keys.py

```diff
@@ -967,7 +1105,8 @@ def blob(self):
                     common.MP(data['q']) + common.MP(data['g']) +
                     common.MP(data['y']))
         else:
```

Comment:
For unknown key types this used to raise an exception. I think it should probably still do that. Let's validate that we have the key type in question in our hands.